File: app/javascript/channels/users_channel.js
import consumer from "./consumer"

export function subscribeToUser(userId)
{consumer.subscriptions.create({ channel: "UsersChannel", user_id: userId}, {
    connected() {
        // Called when the subscription is ready for use on the server
        console.log('Subscribed')
    },

    disconnected() {
        // Called when the subscription has been terminated by the server
    },

    received(data) {
        console.log('Got some data')
        // Called when there's incoming data on the websocket for this channel
        switch(data.type) {
            default:
                console.log(`Unrecognized ${data.type}`)
        }
    }
})}

File: app/channels/users_channel.rb
# frozen_string_literal: true

class UsersChannel < ApplicationCable::Channel
  def subscribed
    stream_for current_user
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def current_user
    @current_user ||= User.find(params[:user_id])
  end
end
